### ISSUE TITLE
Limit the number of list items that can be stored inside an addon's settings

### ISSUE THREAD

**Issue by DNin01** [ScratchAddons/ScratchAddons#5666]
### Why this would be helpful

Lists are one of the biggest offenders to Scratch Addons' storage, which is [limited](https://github.com/ScratchAddons/ScratchAddons/issues/5190). I mean, look at how many items I could add if I wanted to.
![30 list items in the settings of a single addon](https://user-images.githubusercontent.com/106490990/214464709-41df142e-0863-4f7e-bc1f-094acea39452.png)

### My suggestion

If this is becoming a problem, it might be worth considering limiting how many items that the user can add to each list in addon settings and the length of each of them. I was able to prevent SA from saving my settings by filling up lists with items, and if there are people who add many items to the Cloud games or Customizable navigation bar addons, it could happen to them too.

For Customizable navigation bar, you're probably not going to need any more than 8 links, and for Cloud games, I don't know.

### Possible alternatives

- Find a way to compress the data. Cloud games definitely doesn't need the `https://scratch.mit.edu/` at the start of _every_ list item if you're only going to modify the part after that. We could do this in addition to adding a limit, too.
- Related: #5190

**Comment by WorldLanguages** [comment]
Good idea, we could have a default limit of 10 items and allow it to be customized on each addon.json

**Comment by penguinmoose** [comment]
> Good idea, we could have a default limit of 10 items and allow it to be customized on each addon.json

Do we just hide the "plus" icon when the number of items exceeds 10 (or whatever limit is imposed for that addon)? Or should we gray it out?

**Comment by BroJac5246** [comment]
> > > Good idea, we could have a default limit of 10 items and allow it to be customized on each addon.json
> > 
> > Do we just hide the "plus" icon when the number of items exceeds 10 (or whatever limit is imposed for that addon)? Or should we gray it out?
> 
> I was thinking of an alert when you try to add an 11th item, but that works too. Might want a tooltip to let the user know that they can only add up to 10 items in that case.

I think we should grey it out, since that's less obtrusive but still understandable.

**Comment by WorldLanguages** [comment]
Related: #5849
